### PR TITLE
chore: export buildWorkflowDirBundle helper function

### PR DESCRIPTION
**Pull request by thomaswhyyou**
### Description
This PR contains **zero functional changes** in code — we are only moving and reorganizing some code so that we can pull in the cli package as a dependency in the dashboard app ([here](https://github.com/knocklabs/control/pull/2586/files#r1394856342)), and reuse some of the helpers we have in this package.

The reason for this reorganization is explained in more details in [this thread](https://github.com/knocklabs/control/pull/2586/files#r1394842235), but tldr is that we need to isolate the code that can run in a browser and not just in nodejs runtime so we can import and reuse from the cli package in dashboard without having to polyfill all and every nodejs modules. For now, my proposal is the only nodejs module we allow and polyfill is the `path` module. 

There is one other change I made, and is a proposal I'd like to get feedbacks on: I've marked certain files in this repo with `.isomophic` to indicate the code in those files should be able to run in a browser. Without some explicit indication or marking, I think it'd be hard to maintain and make sure the parts of this code base that is running in a browser remains isomorphic. In reality, this just means no native nodejs module imports with the exception of the `path` module. I'm open to other suffix as well, but the idea here is that we should explicitly mark certain files as such. If we agree with this approach, then we should probably add appropriate code comments in those files to explain, or better yet to automatically lint it somehow?

### Tasks
[KNO-4510](https://linear.app/knock/issue/KNO-4510/[cli]-export-buildworkflowdirbundle-helper-from-workflowwriterts)

### CODE REVIEW

**Review by francoborr — COMMENTED** [review]
1) I really like the approach you took here! Reusing our `write` logic for workflows (and in the future layouts or other resources) looks like the way to go.

2) I do agree we must mark which files can run in a browser, but having these `isomorphic` files spread all over the app seems a bit wrong maybe?

What do you think if we have a folder like `src/marshal/isomorphic/...` where all isomorphic files lives? I think it would be easier for us to see and manage all these files + we wouldn't probably need to add comments to all of them explaining what they are.

**Comment by thomaswhyyou** [comment]
@francoborr 
> What do you think if we have a folder like `src/marshal/isomorphic/...` where all isomorphic files lives? I think it would be easier for us to see and manage all these files + we wouldn't probably need to add comments to all of them explaining what they are.

Thanks franco - so this is an approach I considered but ruled against, because I think we have a lot more to lose in terms of the context of what every file is about by doing it this way. 

Basically we have two options for how we can go about organizing the files here:
1) Put all isomorphic files under one directory (let's say src/isomorphic), but the files that go in there are all unrelated. Think back to early days of React, imagine you have src/css/ directory where you put all css files; then each component imports what they need from src/css/, but css files in there are all unrelated to each other. 
2) Keep the files where they are, which is currently organized by the usage and resource context, but mark certain files as "isomorphic". So here it would be like putting css files under each component directory where they are used. With this approach you end up with css files all over the places but each css file is located where it relates to. 

I've taken the approach 2 here, because I thought it feels more intuitive to organize modules contextually.

**Review by meryldakin — APPROVED** [review]
I'm in favor of the way you decided to handle the naming of the isomorphic files. Can we add that explanation into the README?

**Review by connorlindsey — APPROVED** [review]
I like the file naming approach and have used `.server` before. I think it's more in your face which is a good thing here imo especially given accounting for multiple runtimes is somewhat easy to not think of during development. 

Agreed to adding some info in the readme as well

**Review by francoborr — APPROVED** [review]
Gotcha, thanks for the explanation!

**Comment by thomaswhyyou** [comment]
> I'm in favor of the way you decided to handle the naming of the isomorphic files. Can we add that explanation into the README?

Sure, good idea. I was thinking we would add something to every file marked as isomorphic at the top to make it really hard to miss it.

File: .eslintrc.json [hunk 1]
```diff
@@ -14,5 +14,18 @@
...
+      "rules": {
+        "no-restricted-imports": ["error", {
+          "patterns": [{
+            "group": ["node:*", "!node:path"],
+            "message": "See https://github.com/knocklabs/knock-cli/pull/257 for more details."
+          }]
+        }]
+      }
+    }
+  ]
```
Comment (thomaswhyyou) [inline .eslintrc.json:30]: @meryldakin @connorlindsey Btw I realized our README gets auto generated when we build a release so I'm not sure if that's going to work. However, I did figure out how to add this eslint rule to automatically check for any disallowed node module imports in isomorephic files, so I think we will have a good guardrail for this that way. And finally I added notes at the top of each isomorphic files for a bit more context.

**Pull request merged**